api/agent/agent.go

```diff
@@ -205,7 +208,7 @@ func (a *agent) Submit(callI Call) error {
 	defer slot.Close() // notify our slot is free once we're done
 
 	// TODO Start is checking the timer now, we could do it here, too.
-	err = call.Start(ctx)
+	err = call.Start(ctx, a)
```

Comment:
probably just shove these onto `call.Listeners` when we make a call (add new field), to keep `Start`/`End` clean of parameters. can do this later.